plugins/train/trainer/_base.py

```diff
@@ -288,31 +289,28 @@ def train_one_batch(self, do_preview):
         loss = loss if isinstance(loss, list) else [loss]
         return loss
 
-    def get_next(self, do_preview):
+    def get_next(self):
         """ Return the next batch from the generator
-            Items should come out as: (warped, target [, mask]) """
+            Items should come out as: (sample, warped, targets, [mask]) """
+        logger.debug("Generating targets")
         batch = next(self.feed)
-        if self.use_mask:
-            batch = [[batch["feed"], batch["masks"]], batch["targets"] + [batch["masks"]]]
-        else:
-            batch = [batch["feed"], batch["targets"]]
-        self.generate_preview(do_preview)
-        return batch
+        targets_use_mask = self.model.training_opts["replicate_input_mask"]
+        model_inputs = batch["feed"] + batch["masks"] if self.use_mask else batch["feed"]
+        model_targets = batch["targets"] + batch["masks"] if targets_use_mask else batch["targets"]
+        return model_inputs, model_targets
```

Comment:
Clarify the necessary outputs from the generator and the required inputs to train on batch. Confusing to use a single batch variable before